src/main/java/de/tum/in/www1/artemis/service/LectureUnitProcessingService.java

```diff
@@ -0,0 +1,158 @@
+            OutputStream fileItemOutputStream = fileItem.getOutputStream();
+            IOUtils.copy(input, fileItemOutputStream);
+            MultipartFile multipartFile = new CommonsMultipartFile(fileItem);
+
+            LectureUnitDTO lectureUnitsDTO = new LectureUnitDTO(attachmentUnit, attachment, multipartFile);
+            units.add(lectureUnitsDTO);
+            tempFile.deleteOnExit();
+        }
+
+        document.close();
+        return units;
+    }
+
+    /**
+     * Prepare information of split units for client
+     * @param file The file (lecture slide) to be split
+     * @return The prepared information of split units as list of LectureUnitSplitDTO
+     */
+    public LectureUnitInformationDTO getSplitUnitData(MultipartFile file) throws IOException {
+
+        List<LectureUnitSplitDTO> units = new ArrayList<>();
+        PDDocument document = PDDocument.load(file.getBytes());
+        Tuple<Map<Integer, Tuple<String, Tuple<Integer, Integer>>>, Integer> unitsInformation = separateIntoUnits(document);
+        Map<Integer, Tuple<String, Tuple<Integer, Integer>>> unitsDocumentMap = unitsInformation.x();
+        int numberOfPages = unitsInformation.y();
+
+        unitsDocumentMap.forEach((k, v) -> {
+            LectureUnitSplitDTO newLectureUnit = new LectureUnitSplitDTO(v.x(), ZonedDateTime.now(), v.y().x(), v.y().y());
+            units.add(newLectureUnit);
+        });
+
+        document.close();
+        return new LectureUnitInformationDTO(units, numberOfPages);
+    }
+
+    /**
+     * This method prepares a map with information on how the slide
+     * is going to be split. The map looks like the following:
+     * Map<OutlineNumber, (UnitName, (StartPage, EndPage))>
+     *
+     * @param document The document (lecture slide) to be split
+     * @return The prepared map
+     */
+    private Tuple<Map<Integer, Tuple<String, Tuple<Integer, Integer>>>, Integer> separateIntoUnits(PDDocument document) throws IOException {
+        Map<Integer, Tuple<String, Tuple<Integer, Integer>>> outlineMap = new HashMap<>();
+
+        Splitter pdfSplitter = new Splitter();
+        PDFTextStripper pdfStripper = new PDFTextStripper();
+        List<PDDocument> pages = pdfSplitter.split(document);
+
+        ListIterator<PDDocument> iterator = pages.listIterator();
+
+        int outlineCount = 0;
+        int index = 1;
+        while (iterator.hasNext()) {
+            PDDocument pdCurrent = iterator.next();
+            String slideText = pdfStripper.getText(pdCurrent);
+
+            if (slideText.contains("Outline")) {
+                outlineCount++;
+                String[] lines = slideText.split("\r\n|\r|\n");
+                String unitName = lines[outlineCount + 1].replaceAll("[^a-zA-Z0-9\\s()_-]", "").replaceFirst("^\\s*", "");
+                outlineMap.put(outlineCount, new Tuple<>(unitName, new Tuple<>((outlineCount == 1) ? 1 : index, document.getNumberOfPages())));
```

Comment:
Yes its for obtaining unit name. This might be extracted wrong in some edge cases but since users will have options to edit before submitting i didn't want to make it mare complex for the moment.